plugins/history/databasemanager.cpp

```diff
@@ -0,0 +1,140 @@
+		      " `session`, `session_name`, `from`, `from_name`, `to`, `to_name`, `state`, `type`, `is_group`) "
+		      " VALUES (:timestamp, :message, :account, :protocol, :direction, :importance, :contact, :subject, :session, :session_name, "
+		      " :from, :from_name, :to, :to_name, :state, :type, :is_group)");
+
+	//Add the values to the database fields.
+    query.bindValue(":" + DatabaseConstants::columnTimeStamp() , QString::number(message.timestamp().toTime_t()));
+    query.bindValue(":" + DatabaseConstants::columnMessage(), message.parsedBody());
+    query.bindValue(":" + DatabaseConstants::columnAccount(), message.manager()->account()->accountId());
+    query.bindValue(":" + DatabaseConstants::columnProtocol(), message.manager()->account()->protocol()->pluginId());
+    query.bindValue(":" + DatabaseConstants::columnDirection(), QString::number(message.direction()));
+    query.bindValue(":" + DatabaseConstants::columnImportance(), QString::number(message.importance()));
+
+	QString contact;
+	if (message.direction() == Kopete::Message::Outbound) {
+		contact = message.to().at(0)->contactId();
+	} else {
+		contact = message.from()->contactId();
+	}
+    query.bindValue(":" + DatabaseConstants::columnContact(), contact);
+
+    query.bindValue(":" + DatabaseConstants::columnSubject(), message.subject());
+    query.bindValue(":" + DatabaseConstants::columnSession(), "");
+    query.bindValue(":" + DatabaseConstants::columnSessionName(), "");
+    query.bindValue(":" + DatabaseConstants::columnFrom(), message.from()->contactId());
+    query.bindValue(":" + DatabaseConstants::columnFromName(), message.from()->displayName());
+
+	//If we are dealing with only one recepient, we save this as a single user chat
+	if (message.to().count() == 1) {
+        query.bindValue(":" + DatabaseConstants::columnTo(), message.to().at(0)->contactId());
+        query.bindValue(":" + DatabaseConstants::columnToName(), message.to().at(0)->displayName());
+        query.bindValue(":" + DatabaseConstants::columnIsGroup(), "0");
+	} else {
+		//Otherwise we will create a string with the contact ids of the recepients, and another string to
+		//hold the contact names of the recepients. Both these strings are comma delimited.
+		QString to, to_name;
+		foreach (Kopete::Contact *c, message.to()) {
+			to.append(c->contactId() + ",");
+			to_name.append(c->displayName() + ",");
+		}
+		to.chop(1);
+		to_name.chop(1);
+        query.bindValue(":" + DatabaseConstants::columnTo(), to);
+        query.bindValue(":" + DatabaseConstants::columnToName(), to_name);
+        query.bindValue(":" + DatabaseConstants::columnIsGroup(), "1");
+	}
+    query.bindValue(":" + DatabaseConstants::columnState(), QString::number(message.state()));
+    query.bindValue(":" + DatabaseConstants::columnType(), QString::number(message.type()));
+
+	//Save the query to the database.
+	query.exec();
+}
+
+DatabaseManager *DatabaseManager::instance()
```

Comment:
I assume that allocates the instance on the heap memory, right? Do we need to worry about memory deallocation?